### DISCUSSION TITLE
Starter project - UI deps should be owned by the app or the package

### DISCUSSION THREAD

**Discussion by fibonacid** [vercel/turborepo#484] [Help] [answered]
I love the simplicity and speed of sharing ui components using `next-transpile-modules`, although i have a lingering question on my head.

Third party dependencies, who should specify them? the app? the package? both? 😖 

I'm sure this is more a matter of opinion but i'd like to follow convention if there is one.
Specifying the dependencies on both workspaces seems better because it gives you better editor support on VSCode, although it feels a bit redundant and hard to maintain.

Should I specify the ui package dependencies as `peerDependencies` or that would just create more problems?

**Comment by jaredpalmer** [comment, marked as answer]
I _guess_ you could do what you're describing with `peerDependencies`. However, I think it will be easier to simply keep them as `dependencies` of the `ui` package as then you can collocate and encapsulate real dependency needs of the package. Also, if you ever do need to publish the package, then it's ready to go. It's also going to be less work as you add more applications to only need to specify deps for `ui` in one place instead of in each app that consumes it.

**Reply by fibonacid** [reply]
What happens when i need to use something from the same third party library inside another workspace (web, docs, or util package)
If I add it to both packages json then i have to remember to keep them on the same version.
Re-exporting the library from the ui package doesn't feel like a good idea either.
What do you suggest?
